docs/design/design-patterns/non-functional-requirements_investigation_guide.md

```diff
@@ -0,0 +1,148 @@
+# Non-Functional Requirements Investigation
```

Comment:
The filename mixes use of dashes and underscores - other files under `design` prefer dashes, please update to match.